Protacon.RxMq.AzureServiceBusLegacy/Topic/AzureBusTopicSubscriber.cs

```diff
@@ -81,30 +83,42 @@ internal Binding(MessagingFactory messagingFactory, NamespaceManager namespaceMa
             }
 
-            private static void MakeSureSubscriptionExists(NamespaceManager namespaceManager, AzureTopicMqSettings settings,
-                string topicPath, string subscriptionName)
+            private void MakeSureSubscriptionExists(NamespaceManager namespaceManager, AzureTopicMqSettings settings,
+                string topicPath, string subscriptionName, bool removePrevious = false)
             {
+                var subscriptionDescription = new SubscriptionDescription(topicPath, subscriptionName);
                 if (!namespaceManager.SubscriptionExists(topicPath, subscriptionName))
                 {
-                    var subscriptionDescription = new SubscriptionDescription(topicPath, subscriptionName);
                     namespaceManager.CreateSubscription(settings.SubscriptionBuilderConfig(subscriptionDescription, typeof(T)));
+                    _logMessage($"MakeSureSubscriptionExists: Created subscription: {subscriptionName}");
+                }
+                else
+                {
+                    if (removePrevious)
+                    {
+                        namespaceManager.DeleteSubscription(topicPath, subscriptionName);
+                        namespaceManager.CreateSubscription(settings.SubscriptionBuilderConfig(subscriptionDescription, typeof(T)));
+                        _logMessage($"MakeSureSubscriptionExists: Deleted and created subscription: {subscriptionName}");
+                    }
                 }
             }
 
-            private static void MakeSureTopicExists(NamespaceManager namespaceManager, AzureTopicMqSettings settings,
+            private void MakeSureTopicExists(NamespaceManager namespaceManager, AzureTopicMqSettings settings,
                 string topicPath)
             {
                 if (!namespaceManager.TopicExists(topicPath))
                 {
                     var queueDescription = new TopicDescription(topicPath);
                     namespaceManager.CreateTopic(settings.TopicBuilderConfig(queueDescription, typeof(T)));
+                    _logMessage($"MakeSureTopicExists: Created topic: {topicPath}");
                 }
             }
 
             private void OptionsOnExceptionReceived(object sender, ExceptionReceivedEventArgs exceptionEventArgs)
             {
-                _logError($"Action '{exceptionEventArgs.Action}' caused exception {exceptionEventArgs.Exception}.");
-                if (exceptionEventArgs.Exception is MessagingEntityNotFoundException || exceptionEventArgs.Exception is MessagingCommunicationException)
+                bool addErrorAction = exceptionEventArgs.Exception is MessagingEntityNotFoundException || exceptionEventArgs.Exception is MessagingCommunicationException;
+                _logError($"Action '{exceptionEventArgs.Action}' caused exception '{exceptionEventArgs.Exception.GetType()}', added to error actions '{addErrorAction}'. {Environment.NewLine}{exceptionEventArgs.Exception}.");
```

Comment:
what if exception is something else?

Reply:
Currently we dont want to re-create and handle these other exceptions. These two we want to manage and re-create the subscription when these occur. 

I tested to catch EntityNotFound exception also, but it isnt nessacery and caused more problems when active subscriber changes from A to B.